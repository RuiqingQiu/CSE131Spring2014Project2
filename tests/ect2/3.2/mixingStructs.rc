const int c = 10;
structdef S1 {int [c]a; float[c]b;};
structdef S2 {int [c]a; float[c]b;};
structdef S3 {float [c]a; int [c]b;};
int [2*c] a;
int [2*c] b;

function : int main(){
   int i =0;
   S1 s1; S2 s2; S3 s3;
   while (i < c){
      s1.b[i] = s1.a[i] = i * i * i * i;
      i++;
   }

   s2 = *(S2 *)&s1;
   s3 = *(S3 *)&s2;
   s1 = *(S1 *)&s3;
   
   i = 0; while (i < c){
      cout << s1.a[i] << "\t" << s1.b[i] << endl;
      cout << s2.a[i] << "\t" << s2.b[i] << endl;
      cout << s3.a[i] << "\t" << s3.b[i] << endl;
      cout << endl;
   i++;}

   return 0;
}
