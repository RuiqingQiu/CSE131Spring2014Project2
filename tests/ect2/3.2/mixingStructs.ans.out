0	0.00
0	0.00
0.00	0

1	1.00
1	1.00
0.00	1065353216

16	16.00
16	16.00
0.00	1098907648

81	81.00
81	81.00
0.00	1117913088

256	256.00
256	256.00
0.00	1132462080

625	625.00
625	625.00
0.00	1142702080

1296	1296.00
1296	1296.00
0.00	1151467520

2401	2401.00
2401	2401.00
0.00	1159073792

4096	4096.00
4096	4096.00
0.00	1166016512

6561	6561.00
6561	6561.00
0.00	1171064832

